sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -433,6 +433,61 @@ class JDBCSuite extends QueryTest
     assert(ids(2) === 3)
   }
 
+  test("SPARK-34843: columnPartition should generate a correct stride size") {
+
+    val schema = StructType(Seq(
+      StructField("PartitionColumn", DateType, nullable = false)
```

Comment:
Why `nullable = false`?

Reply:
Good catch. No reason to set this. I've taken it out and it's now using the default.